test/cpp/naming/resolver_component_test.cc

```diff
@@ -46,41 +47,83 @@ extern "C" {
 }
 
 using std::vector;
+using testing::UnorderedElementsAreArray;
 
 DEFINE_string(target_name, "", "");
-DEFINE_bool(expect_target_is_balancer, true, "");
 DEFINE_string(expected_addrs, "", "");
 DEFINE_string(expected_chosen_service_config, "", "");
 DEFINE_string(local_dns_server_address, "", "");
+DEFINE_string(expected_lb_policy, "", "");
 
-namespace grpc {
-namespace testing {
+namespace {
 
-static vector<std::string> parse_expected_addrs(const char *expected_addrs) {
-  std::vector<std::string> out;
+class GrpcLBAddress final {
+ public:
+  GrpcLBAddress(std::string address, bool is_balancer)
+      : is_balancer(is_balancer), address(address) {}
 
-  char *p = (char *)expected_addrs;
+  bool operator<(const GrpcLBAddress &other) const {
+    int comparison = this->address.compare(other.address);
+    if (comparison != 0) {
+      return comparison < 0 ? true : false;
+    }
+    return this->is_balancer == false && other.is_balancer == true;
+  }
+
+  bool is_balancer;
+  std::string address;
+};
+
+bool convert_string_to_bool(std::string bool_str) {
```

Comment:
We should probably use C++ style names in this test -- e.g., `ConvertStringToBool()` instead of `convert_string_to_bool()`.